domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2886,67 +2867,65 @@ public void selectStream(String sql, JsonArray params, Handler<AsyncResult<SQLRo
    * @param params  The parameters for the placeholders in sql.
    * @param replyHandler  The query result or the failure.
    */
-  public void selectStream(AsyncResult<SQLConnection> conn, String sql, JsonArray params,
-      Handler<AsyncResult<SQLRowStream>> replyHandler) {
+  public void selectStream(AsyncResult<SQLConnection> conn, String sql, Tuple params,
+                           Handler<AsyncResult<RowStream<Row>>> replyHandler) {
+     selectStream(conn, sql, params, STREAM_GET_DEFAULT_CHUNK_SIZE, replyHandler);
+  }
+
+  void selectStream(AsyncResult<SQLConnection> conn, String sql, Tuple params, int chunkSize,
+      Handler<AsyncResult<RowStream<Row>>> replyHandler) {
     try {
       if (conn.failed()) {
         replyHandler.handle(Future.failedFuture(conn.cause()));
         return;
       }
-      conn.result().queryStreamWithParams(sql, params, replyHandler);
+      final Transaction tx = conn.result().tx;
+      tx.prepare(sql, res -> {
+        if (res.failed()) {
+          log.error(res.cause().getMessage(), res.cause());
+          replyHandler.handle(Future.failedFuture(res.cause()));
+          return;
+        }
+        PreparedQuery pq = res.result();
+        RowStream<Row> rowStream = pq.createStream(chunkSize, params);
+        replyHandler.handle(Future.succeededFuture(rowStream));
+      });
     } catch (Exception e) {
       log.error("select stream sql: " + e.getMessage() + " - " + sql, e);
       replyHandler.handle(Future.failedFuture(e));
     }
   }
 
+    /**
+     * Execute an INSERT, UPDATE or DELETE statement.
+     * @param sql - the sql to run
+     * @param replyHandler - the result handler with UpdateResult
+     */
+  public void execute(String sql, Handler<AsyncResult<RowSet<Row>>> replyHandler)  {
+    execute(sql, Tuple.tuple(), replyHandler);
+  }
+
   /**
-   * Execute an INSERT, UPDATE or DELETE statement.
-   * @param sql - the sql to run
-   * @param replyHandler - the result handler with UpdateResult converted toString().
-   * @deprecated use {@link #execute(String, Handler)} instead.
+   * Get vertx-pg-client connection
+   * @param replyHandler
    */
-  @Deprecated
-  public void mutate(String sql, Handler<AsyncResult<String>> replyHandler) {
-    execute(sql, res -> {
-      if (res.failed()) {
-        replyHandler.handle(Future.failedFuture(res.cause()));
+  public void getConnection(Handler<AsyncResult<PgConnection>> replyHandler) {
+    getClient().getConnection(x -> {
+      if (x.failed()) {
+        replyHandler.handle(Future.failedFuture(x.cause()));
         return;
       }
-      replyHandler.handle(Future.succeededFuture(res.result().toString()));
+     replyHandler.handle(Future.succeededFuture((PgConnection) x.result()));
```

Comment:
It is not very likely but an RMB based module may update or patch vertx-pg-client and then it may return a connection where the casting fails. For better error reporting I suggest:
```suggestion
     try {
       replyHandler.handle(Future.succeededFuture((PgConnection) x.result()));
     } catch (Exception e) {
       replyHandler.handle(Future.failedFuture(e));
     }
```

Reply:
Makes sence